src/resources/views/crud/inc/form_fields_script.blade.php

```diff
@@ -0,0 +1,95 @@
+<script>
+    /**
+     * A front-end representation of a Backpack field, with its main components.
+     *
+     * Makes it dead-simple for the developer to perform the most common
+     * javascript manipulations, and makes it easy to do custom stuff
+     * too, by exposing the main components (name, wrapper, input).
+     */
+    class CrudField {
+        constructor(fieldName) {
+            this.name = fieldName;
+            this.wrapper = $('[bp-field-name="'+ this.name +'"]');
+            this.input = this.wrapper.find("[bp-field-main-input");
+            // if no bp-field-main-input has been declared in the field itself,
+            // assume it's the first input in that wrapper, whatever it is
+            if (this.input.length == 0) {
+                this.input = $('[bp-field-name="'+ this.name +'"] input, [bp-field-name="'+ this.name +'"] textarea, [bp-field-name="'+ this.name +'"] select').first();
+            }
+            this.value = this.input.val();
+        }
+
+        change(closure) {
+            this.input.change(function(event) {
+                var fieldWrapper = $(this).parent('[bp-field-wrapper=true]');
+                var fieldName = fieldWrapper.attr('bp-field-name');
+                var fieldType = fieldWrapper.attr('bp-field-type');
+                var fieldValue = $(this).val();
+
+                // console.log('Changed field ' + fieldName + ' (type '+ fieldType + '), value is now ' + fieldValue);
+                closure(event, fieldValue, fieldName, fieldType);
+            }).change();
```

Comment:
```suggestion
            });
```

Any particular reason why this change is needed here ? I've tested and it seems it's not needed at all? What am I missing  @tabacitu ? 🙃

Reply:
I had a reason. But I can't remember it 😅 Maybe the order in which the scripts are run? To make sure if the API gets run _after_ the custom JS, it still works? Anyway... I've tested without it to... seems to work soo... yeah, let's remove it. Good catch @pxpm !

Reply:
Wait! Found it! Pushed a commit that adds back the `.change()`.

Take a look at the "Top 10 Scenarios" tab. Without the `.change()` some of the scenarios won't work. They will 
- have fields visible when the page is loaded, then hide them (flash of fields before they're hidden)
- leave fields visible when they should be hidden

Reply:
Hey @tabacitu !

I've been testing this with and without the change and I still think that we should **NOT** call the change if the field didn't change.

The issue about "flashing fields" is just what it is, we can show a "blured form" or a total overlay while form is loading, it's just how it is, html > then > js, it happens with wysiwyg editors too, you can see the textarea before they are turned into editors.

About your second "leave fields visisble" I think you are refering to the `custom_type` in the demo, that should only be visible when value of radio equals custom.

I get what you did here and why you did it, but I think we can't fix that problem the way you did, as that will arise other problems. 

With `.change()` there, we init the html field, and call `change` on it (nothing have changed, still no radio selected, but somewhow it changed).

Try adding validation to a field ? Impossible, if you are validating a field for value empty value, when you have an empty input on the page first load you will have the error triggered.

What I propose to fix this is using another event, like `init` instead of `change`, to separate the initial state from the real change event. 

```javascript
crud.field(monsterFields[i]).onChange(function(e, fieldValue, fieldName, fieldType) {

            if(fieldValue.length <= 0 || typeof fieldValue == "undefined") {
                console.log('gimme value') //add error class, alert user, sirene sound
            }
            console.log('Value for field ' + fieldName + ' (type ' + fieldType +') was changed to '+ fieldValue);
        });
```
Add this on demo `onChange()` test script you will see what I mean.

Please let's think about a solution here, change definitely can't be used for this purpose.